src/20ec0_len_5040.c

```diff
@@ -2,28 +2,174 @@
 #include "map.h"
 
 extern s32 D_800A0F40;
+extern PopupMessage D_800A0BC0[32];
+
+void update_merlee_message(PopupMessage* popup);
+void draw_merlee_message(PopupMessage* popup);
+void func_80045BC8(void);
+void func_80045FB4(void);
+
+extern s16 D_80078160[];
+extern s32 D_80078168[];
+
+void func_80045AC0(void) {
+    PopupMessage* popup;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(D_800A0BC0); i++) {
+        popup = &D_800A0BC0[i];
```

Comment:
I'd prefer this, since declaring `popup` outside the for loop implies it'll be used outside it:
```suggestion
    s32 i;

    for (i = 0; i < ARRAY_COUNT(D_800A0BC0); i++) {
        PopupMessage* popup = &D_800A0BC0[i];
```

Reply:
Same goes for all these functions.